src/qml/bitcoin.cpp

```diff
@@ -160,6 +162,194 @@ void setupChainQSettings(QGuiApplication* app, QString chain)
+
+    m_wallet_list_model = new WalletListModel{*m_node, nullptr};
+
+    m_engine->rootContext()->setContextProperty("networkTrafficTower", m_network_traffic_tower);
+    m_engine->rootContext()->setContextProperty("nodeModel", m_node_model);
+    m_engine->rootContext()->setContextProperty("chainModel", m_chain_model);
+    m_engine->rootContext()->setContextProperty("peerTableModel", m_peer_model);
+    m_engine->rootContext()->setContextProperty("peerListModelProxy", m_peer_model_sort_proxy);
+    m_engine->rootContext()->setContextProperty("walletController", m_wallet_controller);
+    m_engine->rootContext()->setContextProperty("walletListModel", m_wallet_list_model);
+
+    m_options_model->setNode(&(*m_node), m_isOnboarded);
+
+    QObject::connect(m_options_model, &OptionsQmlModel::requestedShutdown, m_executor, &InitExecutor::shutdown);
+
+    m_engine->rootContext()->setContextProperty("optionsModel", m_options_model);
+
+    m_node_model->startShutdownPolling();
+
+    return true;
+}
+
+void startNodeAndTransitionSlot() { createNode(*m_app, *m_engine, m_argc, m_argv, gArgs); }
+
+int initializeAndRunApplication(QGuiApplication* app, QQmlApplicationEngine* m_engine) {
+    AppMode app_mode = SetupAppMode();
+
+    // Register the singleton instance for AppMode with the QML engine
+    qmlRegisterSingletonInstance<AppMode>("org.bitcoincore.qt", 1, 0, "AppMode", &app_mode);
+
+    // Register custom QML types
+    qmlRegisterType<BlockClockDial>("org.bitcoincore.qt", 1, 0, "BlockClockDial");
+    qmlRegisterType<LineGraph>("org.bitcoincore.qt", 1, 0, "LineGraph");
+
+    // Load the main QML file
+    m_engine->load(QUrl(QStringLiteral("qrc:///qml/pages/main.qml")));
+
+    // Check if the QML engine failed to load the main QML file
+    if (m_engine->rootObjects().isEmpty()) {
+        return EXIT_FAILURE;
+    }
+
+    // Get the first root object as a QQuickWindow
+    auto window = qobject_cast<QQuickWindow*>(m_engine->rootObjects().first());
+    if (!window) {
+        return EXIT_FAILURE;
+    }
+
+    // Install the custom message handler for qDebug()
+    qInstallMessageHandler(DebugMessageHandler);
+
+    // Log the graphics API in use
+    qInfo() << "Graphics API in use:" << QmlUtil::GraphicsApi(window);
+
+    // Execute the application
+    return qGuiApp->exec();
+}
+
+bool startNode(QGuiApplication& app, QQmlApplicationEngine& engine, int& argc, char* argv[])
+{
+    m_engine = &engine;
+    QScopedPointer<const NetworkStyle> network_style{NetworkStyle::instantiate(Params().GetChainType())};
+    assert(!network_style.isNull());
+    m_engine->addImageProvider(QStringLiteral("images"), new ImageProvider{network_style.data()});
+
+    m_isOnboarded = true;
+
+    m_options_model = new OptionsQmlModel{nullptr, m_isOnboarded};
+    m_engine->rootContext()->setContextProperty("optionsModel", m_options_model);
+
+    // moved this so that the settings.json file is read and parsed before creating the node
```

Comment:
comment could be more general, without the "moved this"
same as https://github.com/bitcoin-core/gui-qml/pull/408#discussion_r1675628865

Reply:
makes sense